File: euler_045.py
from time import time
from math import sqrt


def is_integer(f):
    return f-int(f) == 0.0

def is_pentagonal(i):
    return is_integer((1 + sqrt(1 + 24*i))/6)

def is_hexagonal(i):
    return is_integer((1 + sqrt(1 + 8 * i)) / 4)

def triangular(n):
    return int(n*(n  + 1)/2)

def triple_gen():
    n = 1
    while True:
        t = triangular(n)
        if is_pentagonal(t) and is_hexagonal(t):
            yield (n, t)
        n += 1

start = time()

tg = triple_gen()

print(next(tg))
print(next(tg))
print(next(tg))


stop = time()
print("Time: " + str(stop-start))

File: euler_045.js
const start = Date.now();

function* triangular() {
    let n = 2;
    let t = 3;
    while (true) {
        yield t;
        n++;
        t += n;
    }
}

const tgen = triangular();

const isInteger = (n) => {
    return Math.floor(n) === n;
}

const isPentagonal = (n) => {
    return isInteger((1 + Math.sqrt(1 + 24 * n)) / 6);
}

const isHexagonal = (n) => {
    return isInteger((1 + Math.sqrt(1 + 8 * n)) / 4);
}

let counter = 0;
let res;

while (counter < 2) {
    let t = tgen.next().value;
    if (isPentagonal(t) && isHexagonal(t)) {
        counter++;
        if (counter === 2) {
            res = t;
        }
    }
}


const end = Date.now();

console.log("Execute time: ", end - start, " The next triangle number is: ", res);